core/src/test/java/org/elasticsearch/cluster/ClusterChangedEventTests.java

```diff
@@ -355,21 +358,64 @@ private static RoutingTable createRoutingTable(final long version, final MetaDat
     }
 
     // Create a list of indices to add
-    private static List<String> addIndices(final int numIndices) {
-        final List<String> list = new ArrayList<>();
+    private static List<Index> addIndices(final int numIndices, final String id) {
+        final List<Index> list = new ArrayList<>();
         for (int i = 0; i < numIndices; i++) {
-            list.add("newIdx_" + i);
+            list.add(new Index("newIdx_" + id + "_" + i, Strings.randomBase64UUID()));
         }
         return list;
     }
 
     // Create a list of indices to delete from a list that already belongs to a particular cluster state.
-    private static List<String> delIndices(final int numIndices, final List<String> currIndices) {
-        final List<String> list = new ArrayList<>();
+    private static List<Index> delIndices(final int numIndices, final List<Index> currIndices) {
+        final List<Index> list = new ArrayList<>();
         for (int i = 0; i < numIndices; i++) {
             list.add(currIndices.get(i));
         }
         return list;
     }
 
+    // execute the indices changes test by generating random index additions and deletions and
+    // checking the values on the cluster changed event.
+    private static ClusterState executeIndicesChangesTest(final ClusterState previousState,
+                                                          final TombstoneDeletionQuantity deletionQuantity) {
+        final int numAdd = randomIntBetween(0, 5); // add random # of indices to the next cluster state
+        final List<Index> stateIndices = new ArrayList<>();
+        for (Iterator<IndexMetaData> iter = previousState.metaData().indices().valuesIt(); iter.hasNext();) {
+            stateIndices.add(iter.next().getIndex());
+        }
+        final int numDel;
+        switch (deletionQuantity) {
+            case DELETE_ALL: numDel = stateIndices.size(); break;
+            case DELETE_NONE: numDel = 0; break;
+            case DELETE_RANDOM: numDel = randomIntBetween(0, Math.max(stateIndices.size() - 1, 0)); break;
+            default: throw new IllegalArgumentException("Unhandled mode " + deletionQuantity);
```

Comment:
You could, but then `numDel` won't necessarily be fully initialized (b/c the assert may not be turned on), so the compiler complains, unless we add some dummy `numDel` value for the `default` case, which I don't see as ideal.